src/tactic/rcases.lean

```diff
@@ -6,11 +6,59 @@ Authors: Mario Carneiro
 import data.dlist
 import tactic.core
 
+/-!
+
+# Recursive cases (`rcases`) tactic and related tactics
```

Comment:
This is a very clear explanation and perhaps some of it should end up in the tactic docs for `rcases` etc.